pkg/controllers/subnet/subnet_controller.go

```
@@ -0,0 +1,246 @@
+		return nil
+	}
+	portNums := len(common.ServiceMediator.GetPortsOfSubnet(*nsxSubnets[0].Id))
+	if portNums > 0 {
+		err := errors.New("subnet still attached by port")
+		log.Error(err, "", "ID", *nsxSubnets[0].Id)
+		return err
+	}
+	return r.Service.DeleteSubnet(nsxSubnets[0])
+}
+
+func (r *SubnetReconciler) updateSubnetStatus(obj *v1alpha1.Subnet) error {
+	nsxSubnet := r.Service.SubnetStore.GetByKey(string(obj.GetUID()))
+	obj.Status.IPAddresses = obj.Status.IPAddresses[:0]
+	statusList, err := r.Service.GetSubnetStatus(nsxSubnet)
+	if err != nil {
+		return err
+	}
+	for _, status := range statusList {
+		obj.Status.IPAddresses = append(obj.Status.IPAddresses, *status.NetworkAddress)
+	}
+	obj.Status.NSXResourcePath = *nsxSubnet.Path
+	return nil
+}
+
+func (r *SubnetReconciler) setSubnetReadyStatusTrue(ctx *context.Context, subnet *v1alpha1.Subnet) {
+	newConditions := []v1alpha1.Condition{
+		{
+			Type:    v1alpha1.Ready,
+			Status:  v1.ConditionTrue,
+			Message: "NSX Subnet has been successfully created/updated",
+			Reason:  "SubnetCreated",
+		},
+	}
+	r.updateSubnetStatusConditions(ctx, subnet, newConditions)
+}
+
+func (r *SubnetReconciler) setSubnetReadyStatusFalse(ctx *context.Context, subnet *v1alpha1.Subnet) {
+	newConditions := []v1alpha1.Condition{
+		{
+			Type:    v1alpha1.Ready,
+			Status:  v1.ConditionFalse,
+			Message: "NSX Subnet could not be created/updated",
+			Reason:  "SubnetNotReady",
+		},
+	}
+	r.updateSubnetStatusConditions(ctx, subnet, newConditions)
+}
+
+func (r *SubnetReconciler) updateSubnetStatusConditions(ctx *context.Context, subnet *v1alpha1.Subnet, newConditions []v1alpha1.Condition) {
+	conditionsUpdated := false
+	for i := range newConditions {
+		if r.mergeSubnetStatusCondition(ctx, subnet, &newConditions[i]) {
+			conditionsUpdated = true
+		}
+	}
+	if conditionsUpdated {
+		r.Client.Status().Update(*ctx, subnet)
+		log.V(1).Info("updated Subnet", "Name", subnet.Name, "Namespace", subnet.Namespace,
```

It logged success without checking whether the call succeeds or not, which will cause confusion when troubleshooting. Even if it's best effort and doesn't want to return the error, it should log the result accurately.